### ISSUE TITLE
VelocityAt seems to return Velocity relative to wrong body when patch switches SOI

### ISSUE THREAD

**Issue by ColJay** [KSP-KOS/KOS#2685]
predictions of velocity seem to work very well with nodes/patches on the current orbit, unless an SOI is switched. Example - setting an orbit from Kerbin to the Mun resulting in a hyperbolic encounter, velocityat seems to give the velocityvector in relation to kerbin instead of the Mun if the time ahead is in the mun patch.

Attached code illustrates:

```
local vel_at_tm is SHIP:ORBIT:NEXTPATCHETA + 3000.
local tm is TIME:SECONDS + vel_at_tm.

local pred_vel is velocityat(ship, tm):ORBIT.

print "PREDICTED VELOCITY VECTOR     : " + pred_vel.
print "PREDICTED VELOCITY MAGNITUDE  : " + pred_vel:mag.

warpto(tm).
until kuniverse:timewarp:issettled {
    wait 0.
}
wait until time:seconds >= tm.

print "ACTUAL VELOCITY VECTOR        : " + ship:velocity:orbit.
print "ACTUAL VELOCITY MAGNITUDE     : " + ship:velocity:orbit:mag.

print "CALC MATCHING PREDICTION      : " + (ship:velocity:orbit + mun:orbit:velocity:orbit).
print "CALC MATCHING PREDICTION      : " + (ship:velocity:orbit + mun:orbit:velocity:orbit):mag.
```

**Comment by ColJay** [comment]
Just realised that :

```
print "PREDICTED VELOCITY VEC ADJUSTED   : " + (pred_vel - VELOCITYAT(mun, tm):ORBIT).
print "PREDICTED VELOCITY MAG ADJUSTED   : " + (pred_vel - VELOCITYAT(mun, tm):ORBIT):mag.
```

will give the correct predicted velocity

**Comment by Dunbaratu** [comment]
I'm not sure this is wrong, just needs better documentation.

**Comment by Dunbaratu** [comment]
fixed by PR #2777 but it didn't auto-close.
